src/registrar/views/application.py

```diff
@@ -0,0 +1,421 @@
+        Step.ORGANIZATION_ELECTION: _("Type of organization — Election board"),
+        Step.ORGANIZATION_CONTACT: _("Organization name and mailing address"),
+        Step.AUTHORIZING_OFFICIAL: _("Authorizing official"),
+        Step.CURRENT_SITES: _("Organization website"),
+        Step.DOTGOV_DOMAIN: _(".gov domain"),
+        Step.PURPOSE: _("Purpose of your domain"),
+        Step.YOUR_CONTACT: _("Your contact information"),
+        Step.OTHER_CONTACTS: _("Other contacts for your domain"),
+        Step.SECURITY_EMAIL: _("Security email for public use"),
+        Step.ANYTHING_ELSE: _("Anything else we should know?"),
+        Step.REQUIREMENTS: _(
+            "Requirements for registration and operation of .gov domains"
+        ),
+        Step.REVIEW: _("Review and submit your domain request"),
+    }
+
+    # We can use a dictionary with step names and callables that return booleans
+    # to show or hide particular steps based on the state of the process.
+    WIZARD_CONDITIONS = {
+        Step.ORGANIZATION_FEDERAL: lambda w: w.from_model(
+            "show_organization_federal", False
+        ),
+        Step.ORGANIZATION_ELECTION: lambda w: w.from_model(
+            "show_organization_election", False
+        ),
+    }
+
+    def __init__(self):
+        super().__init__()
+        self.steps = StepsHelper(self)
+        self._application = None  # for caching
+
+    @property
+    def prefix(self):
+        """Namespace the wizard to avoid clashes in session variable names."""
+        # this is a string literal but can be made dynamic if we'd like
+        # users to have multiple applications open for editing simultaneously
+        return "wizard_application"
+
+    @property
+    def application(self) -> DomainApplication:
+        """
+        Attempt to match the current wizard with a DomainApplication.
+
+        Will create an application if none exists.
+        """
+        if self._application:
+            return self._application
+
+        if self.has_pk():
+            id = self.storage["application_id"]
+            try:
+                self._application = DomainApplication.objects.get(
+                    creator=self.request.user,  # type: ignore
+                    pk=id,
+                )
+                return self._application
+            except DomainApplication.DoesNotExist:
+                logger.debug("Application id %s did not have a DomainApplication" % id)
+
+        self._application = DomainApplication.objects.create(
+            creator=self.request.user,  # type: ignore
+        )
+        self.storage["application_id"] = self._application.id
+        return self._application
+
+    @property
+    def storage(self):
+        # marking session as modified on every access
+        # so that updates to nested keys are always saved
+        self.request.session.modified = True
+        if self.prefix not in self.request.session:
```

Comment:
**suggestion (non-blocking):** I'm not sure if it's any better that your explicit version here, but this pattern is Python's `setdefault` method: `return self.request.session.setdefault(self.prefix, {})`.

Reply:
I do prefer idiomatic code; I've updated it to use `setdefault`.